paasta_tools/flink_tools.py

```diff
@@ -166,6 +189,71 @@ def _dashboard_get(cr_name: str, cluster: str, path: str) -> str:
     return response.text
 
 
+def _filter_for_endpoint(json_response: Any, endpoint: str) -> Mapping[str, Any]:
+    """
+    Filter json response to include only a subset of fields.
+    """
+    if endpoint == "config":
+        return {
+            key: value for (key, value) in json_response.items() if key in CONFIG_KEYS
+        }
+    if endpoint == "overview":
+        return {
+            key: value for (key, value) in json_response.items() if key in OVERVIEW_KEYS
+        }
+    if endpoint == "jobs":
+        return json_response
+    if endpoint.startswith("jobs"):
+        return {
+            key: value
+            for (key, value) in json_response.items()
+            if key in JOB_DETAILS_KEYS
+        }
+    return json_response
+
+
+def _get_jm_rest_api_base_url(cr: Mapping[str, Any]) -> str:
+    metadata = cr["metadata"]
+    cluster = metadata["labels"][paasta_prefixed("cluster")]
+    base_url = get_flink_ingress_url_root(cluster)
+
+    # dashboard_url = http://flink-jobmanager-host:port/paasta-service-cr-name
+    dashboard_url = urlparse(metadata["annotations"]["flink.yelp.com/dashboard_url"])
+    service_cr_name = dashboard_url.path
```

Comment:
we could get fancy here and do something like:
```suggestion
    _, _, service_cr_name, *_ = urlparse(metadata["annotations"]["flink.yelp.com/dashboard_url"])
```

but that might be a little too clever to just save a single line so maybe ignore me here :p

Reply:
I like fancy :P